presto-tests/src/test/java/com/facebook/presto/tests/TestGracefulShutdown.java

```diff
@@ -62,35 +65,49 @@ public void shutdown()
         executor.shutdownNow();
     }
 
-    @Test(timeOut = SHUTDOWN_TIMEOUT_MILLIS)
-    public void testShutdown()
-            throws Exception
+    @DataProvider(name = "testServerInfo")
+    public static Object[][] testServerInfo()
     {
-        Map<String, String> properties = ImmutableMap.<String, String>builder()
-                .put("node-scheduler.include-coordinator", "false")
-                .put("shutdown.grace-period", "10s")
-                .build();
+        return new Object[][] {
+                {WORKER, ImmutableMap.<String, String>builder()
+                        .put("node-scheduler.include-coordinator", "false")
+                        .put("shutdown.grace-period", "10s")
+                        .build()},
+                {COORDINATOR, ImmutableMap.<String, String>builder()
+                        .put("node-scheduler.include-coordinator", "false")
+                        .put("shutdown.grace-period", "10s")
+                        .build()},
+                {COORDINATOR, ImmutableMap.<String, String>builder()
+                        .put("node-scheduler.include-coordinator", "true")
+                        .put("shutdown.grace-period", "10s")
+                        .build()}
+        };
+    }
 
+    @Test(timeOut = SHUTDOWN_TIMEOUT_MILLIS, dataProvider = "testServerInfo")
+    public void testShutdown(String serverInstanceType, Map<String, String> properties)
+            throws Exception
+    {
         try (DistributedQueryRunner queryRunner = createQueryRunner(TINY_SESSION, properties)) {
             List<ListenableFuture<?>> queryFutures = new ArrayList<>();
             for (int i = 0; i < 5; i++) {
                 queryFutures.add(executor.submit(() -> queryRunner.execute("SELECT COUNT(*), clerk FROM orders GROUP BY clerk")));
             }
-
-            TestingPrestoServer worker = queryRunner.getServers()
+            boolean isCoordinatorInstance = serverInstanceType.equals(COORDINATOR);
+            TestingPrestoServer testServer = queryRunner.getServers()
                     .stream()
-                    .filter(server -> !server.isCoordinator())
+                    .filter(server -> server.isCoordinator() == isCoordinatorInstance)
                     .findFirst()
                     .get();
+            if (!isCoordinatorInstance) {
+                TaskManager taskManager = testServer.getTaskManager();
 
-            TaskManager taskManager = worker.getTaskManager();
-
-            // wait until tasks show up on the worker
-            while (taskManager.getAllTaskInfo().isEmpty()) {
-                MILLISECONDS.sleep(500);
+                // wait until tasks show up on the worker
```

Comment:
remove worker from the comment as now it's going to be either worker/coordinator.